pkg/apis/istio/v1alpha2/istiocontrolplane_types.proto

```diff
@@ -618,6 +620,46 @@ message Resources {
     map<string, string> requests = 2;
 }
 
+// Mirrors k8s.io.api.core.v1.Probe for unmarshaling
+message ReadinessProbe {
+    ExecAction exec = 1;
+    HTTPGetAction httpGet = 2;
+    TCPSocketAction tcpSocket = 3;
+    int32 initial_delaySeconds = 4;
+    int32 timeout_seconds = 5;
+    int32 period_seconds = 6;
+    int32 success_threshold = 7;
+    int32 failure_threshold = 8;
+}
+
+message ExecAction {
+    repeated string command = 1;
+}
+
+message HTTPGetAction {
+    string path = 1;
+    //k8s.io.apimachinery.pkg.util.intstr.IntOrString port = 2;
+    // Currently the port doesn't support port name because these is a jsonpb unmarshal issue,
+    // see: https://github.com/istio/istio/issues/15366#issuecomment-511304787
+    int32 port = 2;
+    string host = 3;
+    string scheme = 4;
+    repeated HTTPHeader httpHeaders = 5;
+}
+
+message HTTPHeader {
+    string name = 1;
+    string value = 2;
+}
+
+message TCPSocketAction {
+    //k8s.io.apimachinery.pkg.util.intstr.IntOrString port = 1;
```

Comment:
Per my comment in the issue, I think we should patch this in the structs manually. We really want to avoid any deviations from k8s schema.

Reply:
Sure, will do it.

Reply:
Have extended the k8s `IntOrString` type and generated patch to the generated go code, also add some unit test cases in `validation_test.go` to make sure both the `port number` and `port name` can be marshal/unmarshal properly.

Reply:
Please check the updated patch file here: https://github.com/morvencao/operator/blob/br_fix_probe/pkg/apis/istio/v1alpha2/fixup_go_structs.patch